### DISCUSSION TITLE
Adding TS tabs to examples

### DISCUSSION THREAD

**Discussion by alexnguyennz** [withastro/docs#6205] [Ideas]
Is it worth it adding TypeScript examples as additional tabs? For example, with the [addClientDirective API examples](https://docs.astro.build/en/reference/integrations-reference/#addclientdirective-option) (or anywhere else) for people who prefer TS instead of using JsDoc.

Not sure if this is covered elsewhere e.g. in the Astro docs docs.

**Comment by anaxite** [comment]
I have occasionally needed help with TypeScript, so I would find TS examples useful if they were there. However, this is also a big task, and there may be other tools to help with figuring out JS->TS.

**Comment by OliverSpeir** [comment]
We discussed this discussion on talking and docing so to capture some of that here:

1. people like this idea, as Astro is a typescript framework 
2. Some suggested to just upgrade existing code snippets to include annotations when they need and to omit them when types can be inferred 
3. Some suggested adding a section to each page or section where types might be required/ not obvious ( integrations / middleware / content collections ) 
4. There were questions about whether to satisfy strictest or strict 
5. If we add tabs should we default to JS or TS
6. Some wondered if there could be some automation involved 

Vaguely in line with this discussion two slash has come up in past as well and so I figured just to add context this was considered previously and the researched yielded [this issue](https://github.com/twoslashes/twoslash/issues/32)
